libs/aws/pyproject.toml

```
@@ -13,13 +13,21 @@ license = "MIT"
 [tool.poetry.dependencies]
 python = ">=3.9"
 langchain-core = "^0.3.68"
-boto3 = ">=1.39.0"
+boto3 = ">=1.39.7"
 pydantic = ">=2.10.0,<3"
 numpy = [
   { version = "^1", python = "<3.12" },
   { version = ">=1.26.0,<3", python = ">=3.12" },
 ]
 
+[tool.poetry.group.tools]
+optional = true
+
+[tool.poetry.group.tools.dependencies]
+bedrock-agentcore = { version = ">=0.1.0", python = ">=3.10" }
+playwright = ">=1.53.0"
+beautifulsoup4 = ">=4.13.4"
```

Poetry dependency groups are generally for dev usage (ex. in our [CI actions](https://github.com/langchain-ai/langchain-aws/blob/d97e1b3493369261eaa45ca7d37117784c67b659/.github/workflows/_test.yml#L41)) - I wasn't able to find a way to install `tools` directly via `pip`, needed to use this from the project directory:
```
poetry install --with tools
```

What I found did work with `pip` was defining tools under `extras` and moving the versioned dependencies to be discoverable by extras (while still optional):
```suggestion
bedrock-agentcore = { version = ">=0.1.0", python = ">=3.10", optional = true }
playwright = { version = ">=1.53.0", optional = true }
beautifulsoup4 = { version = ">=4.13.4", optional = true }

[tool.poetry.extras]
tools = ["bedrock-agentcore", "playwright", "beautifulsoup4"]
```

Then we can install with:
```
pip install 'langchain-aws[tools]'
```